app/controllers/etds_controller.rb

```diff
@@ -105,45 +84,36 @@ def authenticate
     end
   end
 
-  def set_submission
-    return render status: :bad_request, plain: invalid_xml_message unless valid_xml?
+  def validate_and_set_submission_params # rubocop:disable Metrics/AbcSize
+    input_validator = Partner::SubmissionInputValidator.new(
+      **Hash.from_xml(request.raw_post)&.with_indifferent_access
+    )
 
-    @submission = Submission.find_or_initialize_by(dissertation_id:, title:)
-    @message = @submission.new_record? ? 'created' : 'updated'
-  end
+    input_validator.parse!
 
-  # Translate the incoming XML data to attributes on the Submission model
-  # Exclude reader and registrar actions which are handled separately
-  def submission_attributes # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
```

Comment:
Get the controller out of the business of munging edge data into the shape we want.